src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -360,6 +399,42 @@ class CreateExternalTableAsStatementSegment(BaseSegment):
     )
 
 
+@redshift_dialect.segment()
+class CreateLibraryStatementSegment(BaseSegment):
+    """A `CREATE LIBRARY` statement.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_LIBRARY.html
+    TODO: Region sequence can be either before or after the authorization
+    """
+
+    type = "create_library_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        Sequence(
+            "OR",
+            "REPLACE",
+            optional=True,
+        ),
+        "LIBRARY",
+        Ref("ObjectReferenceSegment"),
+        "LANGUAGE",
+        "PLPYTHONU",
+        "FROM",
+        Ref("QuotedLiteralSegment"),
+        Sequence(
+            Ref("AuthorizationSegment", optional=False),
+            Sequence(
+                "REGION",
+                Ref.keyword("AS", optional=True),
+                Ref("QuotedLiteralSegment"),
+                optional=True,
+            ),
+            optional=True,
+        ),
```

Comment:
We don't nice way of doing this so normally just put in `AnyNumberOf` (which also removes the need for the Optional since it allows 0 instances so is optional by default):

```suggestion
        AnyNumberOf(
            Ref("AuthorizationSegment", optional=False),
            Sequence(
                "REGION",
                Ref.keyword("AS", optional=True),
                Ref("QuotedLiteralSegment"),
                optional=True,
            ),
        ),
```

This allows them to be added in any order. It does however also allow two AuthorizationSegments, or two Regions. So not ideal. But the best we have for now.